### PR TITLE
Set fixed width in `IndexMeta`

### LINKED ISSUE #152: Further improve value storage
Each b+ tree corresponds to a single type, so in the case of fixed width types (basically everything except a string), we don't actually need to store the value's length and instead can figure it out from the type of the field. Note, we still need the data pointer as a disambiguator.

In essence, we should pass some information into the b+ tree node that indicates that it's either fixed width or variable width, and if it's fixed width, how wide it should be. Widths are:
* 1 byte for boolean
* 8 bytes for int64 and float64
* 0 bytes for null

### PR DESCRIPTION
**Pull request by friendlymatthew**
Closes #152 

> [!NOTE]
> Still a draft PR. We'll need to update `node.go` and `bptree.go` to pass the `FixedWidth` information and remove the need of storing `l`

## What this PR contains

1. Allocates 2 bytes in the `IndexMeta` to specify the `FixedWidth` for that given multi page
2. Updated the `ts` reader so it reads the `FixedWidth` upon scanning the metadata, drills down to the `UnmarshalNode` logic, replacing the need to store the length (`l`) in disk

- [x] tested and works via demo

### CODE REVIEW

**Comment by kevmo314** [comment]
Nice work so far! Just a note, it's fine for the js API but for the go API we shouldn't add `fieldWidth` to `UnmarshalBinary` to avoid it from violating the `BinaryUnmarshaller` interface.

This admittedly makes it a little tricky but the way we could get around this is manually setting the field width on the struct before unmarshalling.

```
node := &BPTreeNode{ FieldWidth: ... }
node.UnmarshalBinary(...)
```

**Comment by friendlymatthew** [comment]
> Nice work so far! Just a note, it's fine for the js API but for the go API we shouldn't add `fieldWidth` to `UnmarshalBinary` to avoid it from violating the `BinaryUnmarshaller` interface.
> 
> ```
> node := &BPTreeNode{ FieldWidth: ... }
> node.UnmarshalBinary(...)
> ```

Sounds good. I don't think my existing work violates the `UnmarshalBinary` logic for `node.go`. But I'll find an entry point and drill it down to `BpTreeNode`

File: pkg/appendable/index_file.go [hunk 1]
```diff
@@ -185,10 +185,16 @@ func (i *IndexFile) FindOrCreateIndex(name string, fieldType FieldType) (*btree.
...
 	metadata.FieldType = fieldType
+	fieldWidth := DetermineFieldWidth(fieldType)
+	metadata.FieldWidth = fieldWidth
+
 	buf, err := metadata.MarshalBinary()
 	if err != nil {
 		return nil, fmt.Errorf("failed to marshal metadata: %w", err)
 	}
+
+	next.FixedWidth = fieldWidth
```
Comment (friendlymatthew) [inline pkg/appendable/index_file.go:196]: this should correctly store the `FixedWidth` in `LinkedMetaPage` right?

**Comment by friendlymatthew** [comment]
I'm going to close this -- and start new

**Pull request closed without merging**